File: backend/src/domain/usecases/get_current_user.py
from domain.entities import User
from domain.repositories import IAuthRepository


class GetCurrentUser:
    """Use case for getting current authenticated user."""

    def __init__(self, auth_repository: IAuthRepository):
        self.auth_repository = auth_repository

    async def execute(self, user_id: str) -> User:
        """Execute get current user use case."""
        if not user_id:
            raise ValueError("User ID is required")

        return await self.auth_repository.get_current_user(user_id)

File: frontend/src/domain/usecases/GetCurrentUser.ts
import type { IAuthRepository } from "@repositories/IAuthRepository";
import type { User } from "@entities/User";

export class GetCurrentUser {
  private authRepository: IAuthRepository;

  constructor(authRepository: IAuthRepository) {
    this.authRepository = authRepository;
  }

  async execute(): Promise<User> {
    return await this.authRepository.getCurrentUser();
  }
}
